lib/ex_doc/retriever.ex

```diff
@@ -175,6 +178,37 @@ defmodule ExDoc.Retriever do
     end
   end
 
+  defp apply_nested_module_aliases(group_name, title, []), do: {group_name, title}
+
+  defp apply_nested_module_aliases(group_name, title, aliases) do
+    case truncate_if_aliased(title, aliases) do
+      {nil, title} ->
+        {group_name, title}
+
+      {prefix, unprefixed_title} ->
+        group_name =
+          if group_name == nil do
+            aliases
+            |> Map.get(prefix)
+            |> String.to_atom()
+          else
+            group_name
+          end
+
+        {group_name, unprefixed_title}
+    end
+  end
+
+  defp truncate_if_aliased(title, aliases) do
+    aliases
+    |> Map.keys()
+    |> Enum.find(&String.starts_with?(title, &1))
```

Comment:
I agree, as I hadn't really considered this being used for a non-nested module.

You could argue that if a (non nested) module's name needs to be changed, it should probably just be renamed in the code base, but there are projects that are essentially Elixir libraries/wrappers for some other project/service/company that has a really long name. In those cases, the root module name (i.e. first section) is usually a variation of the project/service/company name and "ex", which will therefore also be long. So it's possible this decision will have to be revisited in the future, but my preference would be to keep the implementation/concept simple initially (i.e. limit the feature to nested modules), and enhancing it later as issues arise in the community at large.

This leads me to my next question: how helpful should the code error handling be with improper configuration? E.g. trying to alias non nested modules, giving improper values for the alias value, etc. Should it just crash (e.g. failing to find a matching function clause due to type mismatch) and assume the user will refer to the documentation to find out the proper configuration? Or should "catch all" clauses be added where appropriate to return some sort of nice error message?

Reply:
> So it's possible this decision will have to be revisited in the future, but my preference would be to keep the implementation/concept simple initially (i.e. limit the feature to nested modules), and enhancing it later as issues arise in the community at large.

Exactly. I think it is simpler to start more conservative and expand as necessary. We have the luxury to afford this as we are still pre-1.0. So here are my suggestions:

  1. Consider only really nested modules in the group. So if you nest `Foo.Bar`, `Foo.Bar.Baz` is part of the group but `Foo.Bar` isn't

  2. As proposed by @whatyouhide, let's not support the `:as` option right now. The list given to `nest_module_aliases` is just a simple list of atoms.

I also wouldn't worry about error handling right now. Let's write assertive code, that expects things to be in the proper format, and we can collect feedback in case the API is confusing.